### PR TITLE
Add docs to show that config_dump flag exits

### PR DESCRIPTION
**Pull request by nyanshak**
* Add documentation and logging to show that config_dump flag will exit the
  process after printing the config.

### CODE REVIEW

**Comment by nyanshak** [comment]
To add a bit more context, I ran into an issue and was trying to debug this in the #core room in Slack. What this looked like to me was osquery pulling config then dying after printing out the config, then the watchdog would respawn the process until the overall process died. There was no logging to show why this happened and the docs for `--config_dump` did not explain why.

I think ultimately the behavior should eventually change to match that of `--tls_dump` which does *not* exit after printing the config. However, the more immediate change was to add documentation of the current behavior to help make sure that other people don't run into the same issue.

**Comment by theopolis** [comment]
If you look at the build errors, they suggest a code-format change (`make format_check`)

```
diff --git a/osquery/config/config.cpp b/osquery/config/config.cpp
index 5c0d88e7..b8f6525e 100644
--- a/osquery/config/config.cpp
+++ b/osquery/config/config.cpp
@@ -70,7 +70,10 @@ CLI_FLAG(bool,
          false,
          "Check the format of an osquery config and exit");
 
-CLI_FLAG(bool, config_dump, false, "Dump the contents of the configuration, then exit");
+CLI_FLAG(bool,
+         config_dump,
+         false,
+         "Dump the contents of the configuration, then exit");
```

File: osquery/config/config.cpp [hunk 1]
```diff
@@ -501,6 +501,7 @@ Status Config::refresh() {
                 content.first.c_str(),
                 content.second.c_str());
       }
+      LOG(INFO) << "Requesting shutdown after dumping config";
```
[the remark this answers is not in the document]
Comment (nyanshak) [reply osquery/config/config.cpp]: I'm happy to "disagree and commit", but I want to explain why I disagree.

I think we should try to make it explicit in the output why osquery exits / crashes, as it makes it much easier to debug when the program exits. Even with the documentation, it's possible that someone could miss it and be confused why osquery is "crashing" on them.

[the remark this answers is not in the document]
Comment (theopolis) [reply osquery/config/config.cpp]: I hear where you're coming from. I'd still press for not having this as `LOG(INFO)`. What do you think of `VLOG(1)`, thus needing `--verbose` and so if someone is in a situation where they suspect crashing and they did not see the new docs they will see the signal when debugging.

[the remark this answers is not in the document]
Comment (nyanshak) [reply osquery/config/config.cpp]: Yes, agreed, I'm not familiar with the logging functions of osquery, but I've pushed a new commit that uses `VLOG(1)` instead. That seems like a better way to do this :)

**Comment by theopolis** [comment]
I'll try to fix the flaky test (which represents an actual bug) so we can unblock the test failures here. You will have to run a `git pull --rebase origin master` when that fix has landed.

**Comment by Smjert** [comment]
@nyanshak The fix for the failing test is in master now, please rebase on master so we can merge.

**Comment by Smjert** [comment]
Failure unrelated, kicked again the build for the failed jobs.

**Pull request merged**